content/kotlin/concepts/data-types/data-types.md

```diff
@@ -0,0 +1,365 @@
+
+### Char
+
+In Kotlin, the `Char` data type represents a primitive data type that can hold any character from the Unicode character set, which includes letters, digits, symbols, and special characters.
+
+Here is an example of how to declare a `Char` variable in Kotlin:
+
+```kotlin
+fun main(){
+    val ch: Char = 'A'
+    print(ch)
+}
+```
+
+The output will be:
+
+```
+A
+```
+
+### Double 
+
+In Kotlin, the `Double` data type represents a 64-bit floating-point number. It is a primitive data type that can hold values with a fractional component, such as 3.14 or 0.5.
+
+Here is an example of how to declare a `Double` variable in Kotlin:
+
+```kotlin 
+fun main(){
+    val a: Double = 3.14
+    val b: Double = 2.71
+    val c: Double = a * b as Double 
+    print(c)
+}
+```
+
+The output of the above code will be:
+
+```
+8.5094
+```
+
+In this example, the c variable will be assigned the value 8.5014, which is the result of 3.14 * 2.71.
+
+The `Double` data type can be used to perform more advanced operations, such as calculating the square root of a number or raising a number to a power in Kotlin. For example:
+
+```kotlin
+
+import kotlin.math.sqrt
+import kotlin.math.pow
+fun main(){
+
+
+val a: Double = 16.0
+val b: Double = sqrt(a)
+val c: Double = a.pow(2)
+
+println("The square root of $a is $b.")
+println("$a raised to the power of 2 is $c.")
+
+}
+```
+
+The output will be:
+
+```
+The square root of 16.0 is 4.0.
+16.0 raised to the power of 2 is 256.0.
+```
+
+### Float
+
+In Kotlin, the `Float` data type represents a 32-bit floating-point number. It is a primitive data type that can hold values with a fractional component, such as 3.14 or 0.5.
+
+Here is an example of how to declare a Float variable in Kotlin:
+
+```kotlin
+fun main(){
+    val a: Float = 3.14f
+    val b: Float = 2.71f
+    val c: Float = a * b as Float
+    print(c)
+
+}
+```
+
+The output will be :
+
+```
+8.5094
+```
+In this example, the c variable will be assigned the value 8.5014, which is the result of 3.14 * 2.71.
+
+### Int
+
+In Kotlin, the `Int` data type represents a 32-bit signed integer. It is a primitive data type that can hold values ranging from -2147483648 to 2147483647.
+
+Here is an example of how to declare an `Int` variable in Kotlin:
+
+```kotlin 
+fun main(){
+    val a: Int = 50
+    val b: Int = 25
+    val c: Int = a + b
+    print(c)
+}
+```
+
+The output of the above code will be:
+```
+75
+```
+
+### Long 
+
+In Kotlin, the `Long` data type represents a 64-bit signed integer. It is a primitive data type that can hold values ranging from -9223372036854775808 to 9223372036854775807.
+
+Here is an example of how to declare a `Long` variable in Kotlin:
+
+```kotlin 
+fun main(){
+    val a: Long = 500000000
+    val b: Long = 250000000
+    val c: Long = a + b
+    print(c)
+}
+```
+The output for the above code will be :
```

Comment:
There need to be line breaks before and after code blocks.

Reply:
yep I fixed them in the rest of the code too